src/App.tsx

```
@@ -1,61 +1,132 @@
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+import React, { useState } from 'react';
+import { Todo } from './types/Todo';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const todosInit = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
+
+function getNewTodoId(todos: Todo[]): number {
+  const maxId = Math.max(...todos.map(x => x.id));
+
+  return maxId + 1;
+}
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosInit);
+
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const addTodo = (todo: Todo) => {
+    const newTodo = {
+      ...todo,
+      id: getNewTodoId(todos),
+    };
+
+    setTodos(currentTodos => [...currentTodos, newTodo]);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title.trim() || !userId) {
+      return;
+    }
+
+    addTodo({
+      id: 0,
+      title,
+      userId,
+      completed: false,
+      user: getUserById(userId),
+    });
+
+    reset();
+  };
+
+  const handleUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(!userId);
```

The logic for hiding the error message is incorrect. State updates in React are asynchronous, so `userId` still holds its old value when this line is executed. According to the requirements, "hide the message immediately after any change of the field with an error". A simple `setHasUserIdError(false)` would be a more direct way to achieve this.